sycl/test/sub_group/load_store.cpp

```diff
@@ -145,6 +145,8 @@ int main() {
     return 0;
   }
   if (Queue.get_device().has_extension("cl_intel_subgroups")) {
+    typedef bool aligned_bool __attribute__((aligned(16)));
```

Comment:
Why alignment here is 16?
Why do you expect sizeof(bool) to be 1 byte? especially on the device side?

Reply:
@romanovvlad 
AFAIK, size of bool is 8 bits. Are you suggesting I use size_of(bool) in the code? The code currently implement 16 and 32. I am adding 8. Moreover, this also depends on the available spirv builtins where we only have 16, 32, and 8. So independently from bool, if spirv implement these 3, I think we should add the 3 to subgroup.hpp.

As for alignment 16, I just did the same as the other types in the test.

Reply:
Sizeof bool: https://stackoverflow.com/questions/4897844/is-sizeofbool-defined-in-the-c-language-standard.
I think it's OK to provide implementation of the builtins for 8 bits types. It seems the SYCL specification requires that sizeof(bool) on host == sizeof(bool) on device, but I can't find it says sizeof(bool) == 1.
Why not use char instead which is guaranteed to 1 byte by the spec?